otx/cli/utils/tests.py

```diff
@@ -517,3 +520,90 @@ def otx_explain_testing(template, root, otx_dir, args):
                 compare_image = get_actmap(compare_image, (w, h))
                 diff = np.sum((compare_image - output_image) ** 2) == 0
                 assert diff == 0, f"saliency map output is not same as the sample one, with {diff}!"
+
+
+def otx_find_testing(otx_dir):
+    # Find all model template
+    command_line = ["otx", "find", "--template", "--root", otx_dir]
+    assert run(command_line).returncode == 0
+
+    # Find command per tasks
+    for task in find_supported_tasks:
+        command_line = ["otx", "find", "--template", "--root", otx_dir, "--task", task]
+        assert run(command_line).returncode == 0
+
+    # Find Backbones per backends
+    for backbone_backends in find_supported_backends:
+        command_line = [
+            "otx",
+            "find",
+            "--backbone",
+            backbone_backends,
+            "--root",
+            otx_dir,
+        ]
+        assert run(command_line).returncode == 0
+
+
+def otx_build_testing(root, otx_dir, args):
+    # Build otx-workspace per tasks check - Default Model Template only
+    for task in build_supported_tasks:
+        command_line = [
+            "otx",
+            "build",
+            "--task",
+            task,
+            "--workspace-root",
+            os.path.join(root, f"otx-workspace-{task}"),
+            "--root",
+            otx_dir,
+        ]
+        assert run(command_line).returncode == 0
+
+    for task, backbone in args.items():
+        task_workspace = os.path.join(root, f"otx-workspace-{task}")
+        # Build Backbone.yaml from backbone type
+        command_line = [
+            "otx",
+            "build",
+            "--backbone",
+            backbone,
+            "--workspace-root",
+            task_workspace,
+            "--root",
+            otx_dir,
+            "--save-backbone-to",
+            os.path.join(task_workspace, "backbone.yaml"),
+        ]
+        assert run(command_line).returncode == 0
+        assert os.path.exists(os.path.join(task_workspace, "backbone.yaml"))
+
+        # Build model.py from backbone.yaml
+        command_line = [
+            "otx",
+            "build",
+            "--model",
+            os.path.join(task_workspace, "model.py"),
+            "--backbone",
+            os.path.join(task_workspace, "backbone.yaml"),
+            "--workspace-root",
+            task_workspace,
+            "--root",
+            otx_dir,
+        ]
+        assert run(command_line).returncode == 0
+
+        # Build model.py from backbone type
+        command_line = [
+            "otx",
+            "build",
+            "--model",
+            os.path.join(task_workspace, "model.py"),
+            "--backbone",
+            backbone,
+            "--workspace-root",
+            task_workspace,
+            "--root",
+            otx_dir,
+        ]
+        assert run(command_line).returncode == 0
```

Comment:
I have two questions for test. 
1. Other tests such as train and evaluation will be run in the workspace built by this build test
If not, should we test other cli tools in the workspace?
2. Are there codes for deleting workspace generated by this build test? 
If not, is there no danger about volume in test server?

Reply:
Thanks for the great question. 
1. This could be added, but the role of `otx build` is to update otx model.py and check till train. So it doesn't check other otx commands. It might be more appropriate to add test code just for the workspace.
2. I confirmed that this is created in the `/tmp/` folder and deleted after the test in my local area.